pkg/asset/installconfig/azure/client.go

```diff
@@ -274,12 +275,28 @@ func (c *Client) GetDiskEncryptionSet(ctx context.Context, subscriptionID, group
 	return &diskEncryptionSet, nil
 }
 
+func (c *Client) GetVMCapabilities(ctx context.Context, instanceType, region string) (map[string]string, error) {
```

Comment:
Can all of this be replaced by the `getHyperVGenerationVersion` function you created? I don't see that function being used anywhere.